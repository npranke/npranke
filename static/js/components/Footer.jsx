function Footer() {
    const currentYear = new Date().getFullYear()

    return (
        <footer className="footer" role="contentinfo">
            <div className="copyright">
                &copy; 2017-{currentYear} Nicole Pranke
            </div>
        </footer>
    )
}

export default Footer
