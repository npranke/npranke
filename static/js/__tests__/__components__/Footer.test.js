import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import React from 'react'

import Footer from '@components/Footer'

Enzyme.configure({ adapter: new Adapter() })

describe('Footer', () => {
    test('contains copyright symbol', () => {
        const footer = shallow(<Footer />)

        expect(
            footer.find('.copyright').text(),
        ).toMatch(/Â©/)
    })

    test('contains copyright year', () => {
        const footer = shallow(<Footer />)

        expect(
            footer.find('.copyright').text(),
        ).toMatch(/2019/)
    })
})

describe('Footer snapshot', () => {
    test('matches snapshot', () => {
        const footer = shallow(<Footer />)

        expect(footer).toMatchSnapshot()
    })
})
