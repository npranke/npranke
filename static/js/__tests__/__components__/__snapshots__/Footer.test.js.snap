// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot matches snapshot 1`] = `
<footer
  className="footer"
  role="contentinfo"
>
  <div
    className="copyright"
  >
    Â© 2017-
    2023
     Nicole Pranke
  </div>
</footer>
`;
